### ISSUE TITLE
CORS?

### ISSUE THREAD

**Issue by byaka** [cenobites/flask-jsonrpc#28]
how to provide 'Access-Control-Allow-Origin' header automatically?
i'm try to use this decorator *http://flask.pocoo.org/snippets/56/* but no.
```
@crossdomain(origin='*', methods=['GET', 'OPTIONS', 'POST'])
@jsonrpc.method('addSession')
def addSession(params):
      ...
```

**Comment by nycholas** [comment]
Hello,

Maybe it's the order the decorator, try:

```python
@jsonrpc.method('addSession')
@crossdomain(origin='*', methods=['GET', 'OPTIONS', 'POST'])
def addSession(params):
      ...
```

Thank you.

**Comment by byaka** [comment]
When i'm change order, i have error when try to connect via python console and jsonrpclib (as serverProxy).
```
AppError                                  Traceback (most recent call last)
<ipython-input-57-e2ed3d81507c> in <module>()
----> 1 jsonrpclib.Server('http://0.0.0.0:8099/statsApi').random()

/usr/local/lib/python2.7/dist-packages/jsonrpclib/jsonrpc.pyc in __call__(self, *args, **kwargs)
    559             return self.__send(self.__name, args)
    560         else:
--> 561             return self.__send(self.__name, kwargs)
    562 
    563     def __getattr__(self, name):

/usr/local/lib/python2.7/dist-packages/jsonrpclib/jsonrpc.pyc in _request(self, methodname, params, rpcid)
    437                         config=self._config)
    438         response = self._run_request(request)
--> 439         check_for_errors(response)
    440         return response['result']
    441 

/usr/local/lib/python2.7/dist-packages/jsonrpclib/jsonrpc.pyc in check_for_errors(result)
   1117                 # Application error
   1118                 data = result['error'].get('data', None)
-> 1119                 raise AppError((code, message, data))
   1120 
   1121         elif isinstance(result['error'], dict) and len(result['error']) == 1:

AppError: (200, u'OtherError: 200 OK', None)
```
and in browser i also don't have "Access-Control-Allow-Origin header" in response.

**Comment by byaka** [comment]
with flask_jsonrpc.proxy.ServiceProxy i have error also
```
{u'error': {u'code': 200,
  u'data': None,
  u'message': u'OtherError: 200 OK',
  u'name': u'OtherError'},
 u'id': u'f42a81d2-aee3-11e4-8295-08606ee7f025',
 u'jsonrpc': u'2.0'}
```

**Comment by byaka** [comment]
i'm understand that problem in
```python
make_response(f(*args, **kwargs))

``` from http://flask.pocoo.org/snippets/56/ decorator. This return a new response, but flask-jsonrpc wait not response, only result of function.
Problem how to transfer headers from another decorator to flask-jsonrpc, if flask-jsonrpc doesn't wait a response object...

**Comment by nycholas** [comment]
I understand, would have to add the decorators list to be processed by the JSONRPC.

A while ago there was a pull request[#21] to accomplish this task. At the time, I thought was very intrusive, but now need to think better.

Thank you.

**Comment by byaka** [comment]
Thx for response, i will waiting

**Comment by nycholas** [comment]
@byaka 

It added support to designers that alter the response.

Look at the example decorator[1] for more details.

**Note**: To use this feature has added a helpers `make_response` (from flask_jsonrpc make_response import).

Fixed in version 0.3.

[1] - https://github.com/cenobites/flask-jsonrpc/blob/v0.3/examples/decorator/decorator.py#L70

Thank you.

**Comment by byaka** [comment]
Nice work!
